### ISSUE TITLE
Oracle: PL/SQL pseudorecord Syntax (:NEW, :OLD, :PARENT) not supported

### ISSUE THREAD

**Issue by saulotoledo** [sqlfluff/sqlfluff#7495]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff currently fails to parse Oracle's pseudorecord syntax. When using `:NEW`, `:OLD`, or `:PARENT` to reference column values within a trigger body, the linter triggers a parsing error.

### Expected Behaviour

SQLFluff should recognize the colon-prefixed correlation names as valid identifiers within the Oracle dialect, specifically when followed by a dot-notation column reference.

### Observed Behaviour

The parser fails at the colon (`:`), indicating it does not expect this transition in the Oracle dialect.

### How to reproduce

1. Create `example.sql`:

```sql
CREATE OR REPLACE TRIGGER example_trigger
BEFORE INSERT OR UPDATE ON example_table
FOR EACH ROW
BEGIN
    -- The following line triggers the parse error
    :NEW.column_name := :OLD.column_name + 1;
END;
```

2. Run linting:

```bash
sqlfluff lint example.sql --dialect oracle
```

### Dialect

oracle

### Version

4.0.4

### Configuration

Standard default configuration using `--dialect oracle`.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
